lib/syntax_tree.rb

```diff
@@ -20,6 +18,18 @@
 require_relative "syntax_tree/visitor/match_visitor"
 require_relative "syntax_tree/visitor/pretty_print_visitor"
 
+require_relative "syntax_tree/parser"
+
+# We rely on Symbol#name being available, which is only available in Ruby 3.0+.
+# In case we're running on an older Ruby version, we polyfill it here.
+unless :+.respond_to?(:name)
+  class Symbol # rubocop:disable Style/Documentation
+    def name
```

Comment:
Hum, that polyfill might trick other gems or code into thinking they are on 3.0. Have you considered using a refinement here? I generally avoid them like the plague, but for unobstrusive polyfill I found them relatively useful.

Reply:
Turns out I stopped using it in as many places as I thought, this is much better now

https://github.com/ruby-syntax-tree/syntax_tree/pull/172